lib/sdf-server/src/server/service/diagram/create_connection.rs

```diff
@@ -67,16 +69,22 @@ pub async fn create_connection(
         }),
     );
 
-    WsEvent::connection_created(
-        &ctx,
-        request.from_component_id,
-        request.to_component_id,
-        request.from_socket_id,
-        request.to_socket_id,
-    )
-    .await?
-    .publish_on_commit(&ctx)
-    .await?;
+    let from_component = Component::get_by_id(&ctx, request.from_component_id).await?;
+    let to_component = Component::get_by_id(&ctx, request.to_component_id).await?;
+    for incoming_connection in to_component.incoming_connections(&ctx).await? {
+        if incoming_connection.to_input_socket_id == request.to_socket_id {
```

Comment:
Shouldn't this make sure that all of the `to_socket_id`, `from_component_id`, and `from_socket_id` match? As it currently is, it looks like it'll generate the `WsEvent` for every edge going to the same `InputSocket` as the original request, regardless of which `Component` or `OutputSocket` is on the source side.